pkg/apps/controller/deployer/deployer_controller.go

```diff
@@ -315,8 +336,12 @@ func (c *DeploymentController) nextStatus(pod *v1.Pod, deployment *v1.Replicatio
 			updatedAnnotations[deployapi.DeployerPodCompletedAtAnnotation] = completedTimestamp.String()
 		}
 		return deployapi.DeploymentStatusFailed
+
+	case v1.PodUnknown:
```

Comment:
Seems redundant. Just use the default case.

Reply:
unless you want to log this as weird

Reply:
I wanted to explicitly state that we know about this case so no one reading the code would doubt if we account for that because that enum might be extended in future

Reply:
funny fact: the PodUnknown is never being set in k8s code, so I guess this serves as a future placeholder :-)

Reply:
So to get this straight, this code serves as a future placeholder for changes in the future.
![past-and-future](https://user-images.githubusercontent.com/3900799/31360565-4fb79d5c-ad4f-11e7-8a42-1726d8cd6e99.png)

Reply:
future generations will face pods in unknown state (quantum containers?)

Reply:
@kargakis @mfojtik here are your quantum pods :) https://bugzilla.redhat.com/show_bug.cgi?id=1539143#c0